taskweaver/planner/planner.py

```diff
@@ -195,7 +206,11 @@ def check_post_validity(post: Post):
             assert post.attachment_list[1].type == "plan", "attachment type is not plan"
             assert post.attachment_list[2].type == "current_plan_step", "attachment type is not current_plan_step"
 
-        llm_output = self.llm_api.chat_completion(chat_history, use_backup_engine=use_back_up_engine)["content"]
+        if self.config.skip_planning and rounds[-1].post_list[-1].send_from == "User":
+            llm_output = json.dumps(self.config.dummy_plan)
+            llm_output = llm_output.replace("{request}", rounds[-1].post_list[-1].message)
```

Comment:
better to do the string replacement before JSON dumps in case the user message contains `"` or other special character to break the JSON syntax without properly escaping them. 

(nit: maybe also to extract the source post into a variable)

Reply:
Just modified in # d46ecfe4b091d4804b293d4b50edf872ecbf074c.
Agree with source post extraction. @ShilinHe , how about this?